### ISSUE TITLE
build fails on Fedora 33

### ISSUE THREAD

**Issue by chipitsine** [hashicorp/vault#10657]
fresh master:

```
[ilia@localhost vault]$ git describe
v1.2.0-2160-g4c7105532
[ilia@localhost vault]$ 
```
1. make bootstrap
2. export PATH=/home/ilia/go/bin:$PATH
3. make dev
```
[ilia@localhost vault]$ make dev
==> Checking that build is using go version >= 1.15.3...
==> Using go version 1.15.6...
==> Removing old directory...
==> Building...
Number of parallel builds: 3

-->     linux/amd64: github.com/hashicorp/vault

1 errors occurred:
--> linux/amd64 error: exit status 2
Stderr: # github.com/hashicorp/vault
panic: runtime error: slice bounds out of range [:3449674467] with capacity 15442644

goroutine 1 [running]:
cmd/internal/goobj2.(*Reader).StringAt(...)
	/usr/lib/golang/src/cmd/internal/goobj2/objfile.go:539
cmd/internal/goobj2.(*Reader).StringRef(0xc003084990, 0xc0001536a5, 0x7fca21f7af7a, 0x2)
	/usr/lib/golang/src/cmd/internal/goobj2/objfile.go:561 +0x112
cmd/internal/goobj2.(*Reader).Pkg(...)
	/usr/lib/golang/src/cmd/internal/goobj2/objfile.go:597
cmd/link/internal/loader.(*Loader).resolve(0xc000115c00, 0xc00203b380, 0x10587fffffbd, 0x9fe9a)
	/usr/lib/golang/src/cmd/link/internal/loader/loader.go:603 +0xe5
cmd/link/internal/loader.Reloc2.Sym(0x7fca222e6188, 0xc00203b380, 0xc000115c00, 0x0, 0x9fe9a)
	/usr/lib/golang/src/cmd/link/internal/loader/loader.go:83 +0x48
cmd/link/internal/ld.(*deadcodePass).flood(0xc00027c140)
	/usr/lib/golang/src/cmd/link/internal/ld/deadcode.go:154 +0x1e9
cmd/link/internal/ld.deadcode(0xc000115880)
	/usr/lib/golang/src/cmd/link/internal/ld/deadcode.go:299 +0x306
cmd/link/internal/ld.Main(0x871880, 0x20, 0x20, 0x1, 0x7, 0x10, 0x0, 0x0, 0x6dbb49, 0x1b, ...)
	/usr/lib/golang/src/cmd/link/internal/ld/main.go:235 +0xd3c
main.main()
	/usr/lib/golang/src/cmd/link/main.go:68 +0x1dc

make: *** [Makefile:37: dev] Error 1
[ilia@localhost vault]$ 
```

**Comment by ncabatoff** [comment]
That error is coming from Go's linker.  I don't think it's likely this is anything to do with Vault.  More likely there's something wrong with your build or installation of Go 1.15.6.  Or maybe it's https://github.com/golang/go/issues/41621 ?

**Comment by chipitsine** [comment]
issue mentions `go1.15.2` and issue is resolved, I'm running 1.15.6

btw, does vault running CI on fedora rawhide ?

**Comment by ncabatoff** [comment]
The resolved issue has a backport to 1.15 (https://github.com/golang/go/issues/43214) targeting 1.15.7, currently unreleased.

No, we do not use Fedora in our CI.

**Comment by chipitsine** [comment]
thank you, I'll wait for 1.15.7
